apps/web/src/ui/swap/twap.tsx

```diff
@@ -0,0 +1,237 @@
+import { SimpleSwapHeader } from "./simple/simple-swap-header";
+import { SimpleSwapBridgeBanner } from "./simple/simple-swap-bridge-banner";
+
+const LOGO = {
+  [ChainId.BASE]: "ethereum.svg",
+  [ChainId.ARBITRUM]: "ethereum.svg",
+};
+
+const Modal = ({
+  open,
+  onClose,
+  title,
+  children,
+  header,
+}: {
+  open: boolean;
+  onClose: () => void;
+  title?: string;
+  children?: ReactNode;
+  header?: ReactNode;
+}) => {
+  return (
+    <Dialog open={open} onOpenChange={() => onClose()}>
+      <DialogContent>
+        {" "}
+        <DialogHeader>
+          {header}
+          {title && <DialogTitle>{title}</DialogTitle>}
+        </DialogHeader>
+        {children}
+      </DialogContent>
+    </Dialog>
+  );
+};
+
+const TokenSelectModal = ({
+  selected,
+  children,
+  onSelect,
+}: {
+  selected: any;
+  children: ReactNode;
+  onSelect: (currency: Type) => void;
+}) => {
+  const chainId = useDerivedStateSimpleSwap().state.chainId;
+
+  return (
+    <TokenSelector
+      id={`twap-token-selector`}
+      selected={selected}
+      chainId={chainId}
+      onSelect={onSelect}
+      includeNative={true}
+      hidePinnedTokens={false}
+      hideSearch={false}
+    >
+      {children}
+    </TokenSelector>
+  );
+};
+
+export const useProvider = () => {
+  const [provider, setProvider] = useState<any>(undefined);
+  const { connector, address } = useAccount();
+  const chainId = useChainId();
+
+  const setProviderFromConnector = useCallback(async () => {
+    setProvider(undefined);
+    try {
+      const res = await connector?.getProvider();
+      setProvider(res);
+    } catch (error) {}
+  }, [setProvider, connector, chainId, address]);
+
+  useEffect(() => {
+    setProviderFromConnector();
+  }, [setProviderFromConnector]);
+
+  return provider;
+};
+
+const useTrade = () => {
+  const { data: trade } = useSimpleSwapTrade();
+  const outAmount = trade?.amountOut?.quotient.toString();
+  return {
+    isLoading: !outAmount,
+    outAmount: outAmount,
+  };
+};
+
+const usePriceUSD = (address?: string) => {
+  const chainId = useDerivedStateSimpleSwap().state.chainId;
+
+  const { data: price } = usePrice({
+    chainId,
+    address,
+    enabled: true,
+  });
+
+  return price?.toSignificant(6);
+};
+
+const getTokenLogo = (currency: Currency) => {
+  try {
+    const src = currency.isNative
+      ? `native-currency/${LOGO[currency.chainId as keyof typeof LOGO]}`
+      : `tokens/${currency.chainId}/${currency.wrapped.address}.jpg`;
+    const params = ["f_auto", "c_limit", `w_${64}`];
+    return `https://cdn.sushi.com/image/upload/${params.join(
+      ","
+    )}/d_unknown.png/${src}`;
+  } catch (error) {
+    return "";
+  }
+};
+
+const useTokenList = () => {
+  const { data: customTokenMap } = useCustomTokens();
+  const chainId = useDerivedStateSimpleSwap().state.chainId;
+
+  const { data: otherTokenMap } = useOtherTokenListsQuery({
+    chainId,
+    query: "",
```

Comment:
ideally `query` should be optional on `useOtherTokenListsQuery`, but it's not, think `undefined` would make more sense than `""`